src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/CachedServerInfo.cs

```diff
@@ -108,13 +123,45 @@ public static void AddOrUpdateIsAzure(string  dataSource, bool isAzure)
                         }
                     }
 
-                    info.IsAzure = isAzure;
+                    if (cacheVar == CacheVariable.IsSqlDw)
+                    {
+                        info.IsSqlDw = newState;
+                    }
+                    else if (cacheVar == CacheVariable.IsAzure)
+                    {
+                        info.IsAzure = newState;
+                    }
                     info.LastUpdate = DateTime.UtcNow;
                     _cache.AddOrUpdate(dataSource, info, (key, oldValue) => info);
                 }
             }
         }
 
+        public static void TryGetIsSqlDw(IDbConnection connection, out bool isSqlDw)
+        {
+            Validate.IsNotNull(nameof(connection), connection);
+
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(connection.ConnectionString);
+            TryGetIsSqlDw(builder.DataSource, out isSqlDw);
+        }
+
+        public static void TryGetIsSqlDw(string dataSource, out bool isSqlDw)
```

Comment:
TryGet pattern: return bool indicating if found or not. Do not throw an exception. In references to the code, use `if (!TryGetIsSqlDw(...)` and in the if clause read the value and update the cache if it wasn't already set.